### ISSUE TITLE
Training Loss Become NaN

### ISSUE THREAD

**Issue by 1csu** [jwyang/faster-rcnn.pytorch#247]
When the training process start,the result is shown as follows:
[session 1][epoch  1][iter    0/17998] loss: 5.0623, lr: 1.00e-03
			fg/bg=(8/120), time cost: 5.928952
			rpn_cls: 0.7288, rpn_box: 0.0196, rcnn_cls: 4.1515, rcnn_box 0.1625
[session 1][epoch  1][iter  100/17998] loss: 1.1439, lr: 1.00e-03
			fg/bg=(11/117), time cost: 61.946244
			rpn_cls: 0.2249, rpn_box: 0.1310, rcnn_cls: 0.4164, rcnn_box 0.1736
[session 1][epoch  1][iter  200/17998] loss: nan, lr: 1.00e-03
			fg/bg=(128/0), time cost: 61.668745
			rpn_cls: nan, rpn_box: nan, rcnn_cls: nan, rcnn_box nan
[session 1][epoch  1][iter  300/17998] loss: nan, lr: 1.00e-03
			fg/bg=(128/0), time cost: 63.550695
			rpn_cls: nan, rpn_box: nan, rcnn_cls: nan, rcnn_box nan
[session 1][epoch  1][iter  400/17998] loss: nan, lr: 1.00e-03
			fg/bg=(128/0), time cost: 64.200511
			rpn_cls: nan, rpn_box: nan, rcnn_cls: nan, rcnn_box nan
[session 1][epoch  1][iter  500/17998] loss: nan, lr: 1.00e-03
			fg/bg=(128/0), time cost: 62.974401
			rpn_cls: nan, rpn_box: nan, rcnn_cls: nan, rcnn_box nan
[session 1][epoch  1][iter  600/17998] loss: nan, lr: 1.00e-03
			fg/bg=(128/0), time cost: 64.718690
			rpn_cls: nan, rpn_box: nan, rcnn_cls: nan, rcnn_box nan
I know many people encounter such problem,anyone who finds the solution to this worrying problem can give me some advice?Thanks a lot.

**Comment by dtidmarsh** [comment]
Hi, are you using a custom dataset? I was getting NaN for my custom dataset and I had to increase the anchor sizes because my images were much larger than the default images.

**Comment by 1csu** [comment]
Yes,the default anchor sizes are [8,16,32],what modifications do you modify?

**Comment by 1csu** [comment]
@dtidmarsh .Yes,the default anchor sizes are [8,16,32],what modifications do you modify?

**Comment by 1csu** [comment]
@dtidmarsh  My images are roughly 960x1280 pixels in size, and I modified the code to read:
args.set_cfgs = ['ANCHOR_SCALES', '[16, 32, 64, 128, 256,512]']
However,the Nan issue still exists.I feel very confused.

**Comment by dtidmarsh** [comment]
@1csu If the NaN is a result of bad anchor scales, then you will probably need to experiment a little to find the right scales. Those scales may be too large for the images you have.

**Comment by ly19965** [comment]
@1csu I have a same problem like yours,if you change lr from 1e-3 to 1e-4 ,the loss will be not nan,but the mAP will be 56 ,besides ,if you delete the batchnorm layer in lib/models/fast-rcnn/resnet.py ,lr equals to 1e-3 will be trained well ,but mAP is also bad,almost 30.but it can work well.Therefore, it has problem in batchnorm layer.if you can make it well ,please send email to 987752424@qq.com,thanks.
